common/src/main/java/io/netty/util/concurrent/PromiseTask.java

```diff
@@ -45,7 +45,26 @@ public String toString() {
         }
     }
 
-    protected final Callable<V> task;
+    private static final Callable<?> COMPLETED = new SentinelCallable<Object>("COMPLETED");
+    private static final Callable<?> CANCELLED = new SentinelCallable<Object>("CANCELLED");
+    private static final Callable<?> FAILED = new SentinelCallable<Object>("FAILED");
```

Comment:
can you explain why we even need different instances of this and can not just use the same for all ?

Reply:
The only reason is to make `toString` a bit more useful since it will no longer delegate to the `task.toString()` (see `PromiseTask#toStringBuilder()`). I don't mind either way though ... let me know which you prefer!

Reply:
I see... fine then